keps/sig-scheduling/4815-dra-partitionable-devices/README.md

```diff
@@ -2394,6 +2696,78 @@ mixins:
 The flattened version of this example can be found
 [here](https://gist.github.com/mortent/ddda505e2499c872549fa831dd2459c4).
 
+### Using DRA for the multi-host use-case
+
+In order to allocate a 2x4 TPU slice using the ResourceSlice
+[shown above](#defining-multi-host-devices), a ResourceClaim like the
+following can be used:
+
+```yaml
+apiVersion: resource.k8s.io/v1beta1
+kind: ResourceClaim
+metadata:
+  name: tpu-consumer-resource-claim
+spec:
+  devices:
+    requests:
+    - name: tpu-request
+      deviceClassName: tpu.google.com
+      selectors:
+      - cel:
+          expression: 'device.capacity["tpu.google.com"].tpus == quantity("8")'
+```
+
+This simply requests a device with 8 TPUs. Since there are 4 TPUs per node, this requires
+two pods, one for each node. A Deployment can be used to create the necessary number of
+pods:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: tpu-consumer
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: tpu-consumer
+  template:
+    metadata:
+      labels:
+        app: tpu-consumer
+    spec:
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - weight: 100
+            podAffinityTerm:
+              labelSelector:
+                matchLabels:
+                  app: tpu-consumer
+              topologyKey: kubernetes.io/hostname
+      resourceClaims:
+      - name: "tpu"
+        resourceClaimName: tpu-consumer-resource-claim
```

Comment:
How this should work if a set of pods like in this example will share two or more multi-host resource claims? As I understand for the first pod scheduler can make sure that the node which is selected for a given pod belongs to all selected devices but the following pods will simply get node selectors from the first pod which as I could imagine can lead to situations that those will not be able schedule if the node selectors coming from different devices / resource claims are partially disjoined.

Reply:
Yeah, referencing multiple claims means that there is a chance that either the claims can't be satisfied for any node,  in which case no pods will be scheduled, or it can not be met for all pods, which means that only a subset of the pods will get scheduled. 

The DRA scheduler plugin will generate a node selector based on the selectors from either the ResourceSlice or the device (from the field added in this PR) from all claims. it is possible that this ends up selecting no nodes.

I think that is the expected behavior, as this means the claims can't be satisfied. Obviously it would be better if we could guarantee that no pods got scheduled in this situation, but that is out of scope for DRA (at least at this point). Tools like Kueue can probably help here.